lwcomponent/catalog.go

```diff
@@ -279,63 +278,69 @@ func NewCatalog(
 			}
 		}
 
-		api := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		apiInfo := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		cdkComponents[c.Name] = NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), apiInfo, nil)
+	}
 
-		host, found := localComponents[c.Name]
-		if found {
-			delete(localComponents, c.Name)
-		}
+	components, err := LoadComponents(cdkComponents)
+	if err != nil {
+		return nil, err
+	}
 
-		component := NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), api, host)
+	return &Catalog{client, components, stageConstructor}, nil
+}
 
-		cdkComponents[c.Name] = component
+func NewCachedCatalog(
+	client *api.Client,
+	stageConstructor StageConstructor,
+	cachedComponentsApiInfo map[string]ApiInfo,
+) (*Catalog, error) {
+	if stageConstructor == nil {
+		return nil, errors.New("nil Catalog StageConstructor")
```

Comment:
This error is not helpful to the user 🤔 what should the user do in this case?¿

Reply:
Updated to "StageConstructor is not specified to create new catalog". Not exactly sure what else to say and why we inject stage constructor here. Will there be other types of stagers? cc @jon-stewart